pinot-common/src/main/java/org/apache/pinot/annotations/InterfaceStability.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.annotations;
+import java.lang.annotation.Documented;
+
+/**
+ * Annotation to inform users of how much to rely on a particular package,
+ * class or method not changing over time. Currently the stability can be
+ * {@link org.apache.pinot.annotations.InterfaceStability.Stable},
+ * {@link org.apache.pinot.annotations.InterfaceStability.Evolving} or
+ * {@link org.apache.pinot.annotations.InterfaceStability.Unstable}. <br>
+ *
+ * <ul><li>All classes that are annotated with {@link org.apache.pinot.annotations.InterfaceAudience.Public} or
+ * {@link org.apache.pinot.annotations.InterfaceAudience.LimitedPrivate} must have InterfaceStability annotation. </li>
+ * <li>Classes that are {@link org.apache.pinot.annotations.InterfaceAudience.Private} are to be considered unstable unless
+ * a different InterfaceStability annotation states otherwise.</li>
```

Comment:
Can you think of a use case where we may have a private interface and mark it 'stable'?

Reply:
This wouldn't be typical, but may be possible where internal to pinot code, if a contributor wants to indicate the status of the stability of a class they would have an option to annotate it so.